libs/langchain/langchain/retrievers/google_cloud_enterprise_search.py

```diff
@@ -190,12 +229,25 @@ def _create_search_request(self, query: str) -> SearchRequest:
             spell_correction_spec=spell_correction_spec,
         )
 
+        
+        return response
+
     def _get_relevant_documents(
         self, query: str, *, run_manager: CallbackManagerForRetrieverRun
     ) -> List[Document]:
         """Get documents relevant for a query."""
         search_request = self._create_search_request(query)
-        response = self._client.search(search_request)
-        documents = self._convert_search_response(response.results)
 
-        return documents
+        try:
+            response = self._client.search(search_request)
+        except InvalidArgument as e:
+            raise type(e)(e.message + ' This might be due to engine_data_type not set correctly.' )
+
+        if self.engine_data_type == 0:
+            documents = self._convert_unstructured_search_response(response.results)
+        elif self.engine_data_type == 1:
+            documents = self._convert_structured_search_response(response.results)
+        else:
+            documents = []
```

Comment:
Should the else branch raise an exception saying unsupported engine data type

Reply:
There is another type for website data, I added a comment suggesting to add that into the options

Reply:
Yes exactly - I have added this as a placeholder for when Website will be implemented

Reply:
Could you raise a NotImeplementedError() for the place holder? otherwise this could be a frustrating debugging experience for users with an empty list of docs being returned

Reply:
Until the extra functionality for type website is available , I will replace with ValueError for engine_data_type